### DISCUSSION TITLE
Download Questions / Suggestions

### DISCUSSION THREAD

**Discussion by SFOTyke** [jlai/fitness-dashboard#85] [Ideas]
Firstly, this is a great effort to replace a much loved feature from Fitbit, which was taken away without warning by Google.
I used to download my data on a monthly basis to enter it into a spreadsheet for further manipulation. I can't quite remember exactly, but I believe I selected the Activities and Sleep boxes, which would create a single spreadsheet, with two tabs, one for each of these collections of readings.
Activities data was: Calories Burned, Steps, Distance, Floors, Minutes Sedentary, Minutes Lightly Active, Minutes Fairly Active, Minutes Very Active and Activity Calories.
Sleep data was: Minutes Asleep, Minutes Awake, Number of Awakenings, Time in Bed, Minutes REM Sleep, Minutes Light Sleep and Minutes Deep Sleep. (I had to add the Sleep Score manually to my spreadsheet, as it wasn't downloaded).
I see from the Dashboard that much of this information is available, and some of it is downloadable, albeit in individual datapoints.
Questions:
1. Are there plans to make all these datapoints downloadable?
2. Are there plans to introduce an option to download multiple datapoints for a date range in a single operation?
I realize it is early days and only Beta, but it is a tremendous start.
Thank You everyone!

**Comment by jlai** [comment]
I'm currently not planning on adding a bulk download feature, since making a lot of requests to the API would cause it to exceed the rate limit. It's also a bit iffy from a terms of service point of view.

The official way to export/dump your account data is with https://www.fitbit.com/settings (there should also be a way to get there from the iOS/Android app) although you'll need to convert it into a usable format, e.g. using https://iccir919.github.io/fitbit-json-to-csv/index.html

**Comment by SFOTyke** [comment]
Thanks @jlai for the response and explanation.

I'll take another look at the generic Google Takeout, but previously I could not make sense of the download. I could not identify individual datapoints for individual dates.

I wish I could revert to my old Fitbit login.

**Comment by SFOTyke** [comment]
Hi @jlai,
I was just looking at the Sleep Logs dashboard and I see that I can 4 of the datapoints I need, and calculate another 2.
However, is it possible to add the Number of Awakenings and Sleep Score to this dashboard?
Instead of populating my spreadsheet with a monthly download, I can then update it on a daily basis from your dashboard.
Thanks,
Mal

**Reply by jlai** [reply]
Fitbit doesn't make Sleep Score available via the API so we can't display it.

I've added the wake count to the hover tooltip on the sleep list. Does that help?

**Reply by SFOTyke** [reply]
Hi Jason,
I see you've added the counts to all sleep activities, so going further than I requested. Thank you.
I previously had to add the score manually to my spreadsheet using the iPhone App, which explains why it was not previously part of the Fitbit download. I will continue to do so.
Thanks again.
Mal

**Reply by SFOTyke** [reply]
Hi Jason,
I think I've worked out my Sleep spreadsheet by manually taking data from your dashboard and adding a few calculations for the other fields.
I'm now looking at my Activity spreadsheet. As stated initially above, the Fitbit download included:
**Calories Burned, Steps, Distance, Floors,** Minutes Sedentary, Minutes Lightly Active, Minutes Fairly Active, Minutes Very Active and Activity Calories.
I can find the first four datapoints on the dashboard, but are the others five available, as I could not find them?
Thanks.

**Reply by jlai** [reply]
While the sedentary/lightly/fairly/very active stats are available in the API, the Fitbit app hasn't used them in years (replaced by HR zones) so I'm hesitant to add them as they could be removed in the future.

**Reply by SFOTyke** [reply]
Interesting. The last download I made was in September, before I migrated my User ID from Fitbit to Google, and therefore lost access to the download feature. Those 5 Activity stats were still available in the download. However, I understand what you're saying, as they don't appear to be accessible in the App.

**Reply by SFOTyke** [reply]
Hi @jlai,

As the fields I previously used are not available, I have a question about the new values.
What is the difference between Active Zone Minutes and Heart Rate Zones?
Both appear to have Fat Burn, Cardio and Peak subdivisions, and similar minutes per day, but they do differ slightly.

EDIT: Also, Active Minutes, available in the App and Dashboard, also appears to be a different reading.

Thanks,
Mal

**Reply by jlai** [reply]
Active Zone Minutes counts extra "minutes" for cardio and peak minutes. 1 fat burn minute = 1 AZM, 1 cardio/peak minute = 2 AZM. Active Minutes are just the number of minutes where your activity was over a certain level, without a multiplier. The threshold might be different from the AZM calculation but I'm not sure.

Newer devices support AZM while older ones only support Active minutes.

See https://support.google.com/fitbit/answer/14236509
